### ISSUE TITLE
P3D in processing 3.1 and also 3.0 constant crashing

### ISSUE THREAD

**Issue by ashleyjamesbrown** [processing/processing#4456]
Consistently crashing when attempting to choose P3D as the renderer in the size() command.

Max OSC 10.9.5 Mavericks.
Macbook Pro 2011. Intel Graphics 3000 512MB
Java version 1.8.0_91
Processing Memory set to 512MB

Console Log from Processing attached.

```
0.009 0.0028 0.0
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00001234002e318b, pid=4276, tid=57611
#
# JRE version: Java(TM) SE Runtime Environment (8.0_51-b16) (build 1.8.0_51-b16)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.51-b03 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [AppleIntelHD3000GraphicsGLDriver+0x2e318b]  gldBlitFramebufferData+0x3f8
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/ashleybrown/hs_err_pid4276.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
Could not run the sketch (Target VM failed to initialize).
Make sure that you haven't set the maximum available memory too high.
For more information, read revisions.txt and Help → Troubleshooting.
```

**Comment by ashleyjamesbrown** [comment]
I should add that P2D does the exact same thing.

**Comment by ashleyjamesbrown** [comment]
This runs.

`void setup(){
 size(800,600,P3D);
 background(0);
}


void draw(){
 
}


`


This does not


`void setup(){
 size(800,600,P3D);
 background(0);
}


void draw(){
 ellipse(400,300,100,100);
}`

**Comment by JakubValtar** [comment]
I think this is duplicate of https://github.com/processing/processing/issues/4104 (i.e. dreaded HD3000)

**Comment by ashleyjamesbrown** [comment]
damn you closed it - wanted to say that the new java update got applied and then after a reboot and also updating to 3.1 again and clearing out the old version off my machine seemed to fix it.

Its now running p3d sketches fine.

**Comment by JakubValtar** [comment]
Hi @ashleyjamesbrown, thanks for the report. The thing that we closed this as duplicate does not mean we don't want to hear from you... it means that there is already same issue and we don't want to have multiple issues which are about the same problem, so you should go there to post updates and new information.

Could you please repost your solution to the https://github.com/processing/processing/issues/4104? If you managed to solve it it is awesome news and we want other people with the same problem to try the solution too.

**Comment by Garrettlynch** [comment]
It looks like this error still persists in Processing 3.5.4 and 4.0 on OSX 10.14.6. Reverting back to 2.2.1 resolves the issue.

Specs:

Max OSC 10.14.6 Mojave.
Macbook Pro 2019. 2.6GHz Intel Core i7
Java version 1.8.0_261
Processing Memory set to 512MB


Processing console log:

# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x00007fff395d8b1b, pid=640, tid=0x0000000000001803
#
# JRE version: Java(TM) SE Runtime Environment (8.0_202-b08) (build 1.8.0_202-b08)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.202-b08 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [CoreFoundation+0xab1b]  CFEqual+0x293
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/garrettlynch/hs_err_pid640.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
Could not run the sketch (Target VM failed to initialize).
Make sure that you haven't set the maximum available memory too high.
For more information, read revisions.txt and Help → Troubleshooting.
